tooling/image-sync/internal/repository.go

```diff
@@ -0,0 +1,379 @@
+package internal
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/containers/azcontainerregistry"
+)
+
+// Registry is the interface for accessing image repositories
+type Registry interface {
+	GetTags(context.Context, string) ([]string, error)
+}
+
+// AuthedTransport is a http.RoundTripper that adds an Authorization header
+type AuthedTransport struct {
+	Key     string
+	Wrapped http.RoundTripper
+}
+
+// RoundTrip implements http.RoundTripper and sets Authorization header
+func (t *AuthedTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Set("Authorization", t.Key)
+	return t.Wrapped.RoundTrip(req)
+}
+
+// QuayRegistry implements Quay Repository access
+type QuayRegistry struct {
+	httpclient   *http.Client
+	baseUrl      string
+	numberOftags int
+}
+
+// NewQuayRegistry creates a new QuayRegistry access client
+func NewQuayRegistry(cfg *SyncConfig, bearerToken string) *QuayRegistry {
+	q := &QuayRegistry{
+		httpclient: &http.Client{Timeout: time.Duration(cfg.RequestTimeout) * time.Second,
+			Transport: &AuthedTransport{
+				Key:     "Bearer " + bearerToken,
+				Wrapped: http.DefaultTransport,
+			},
+		},
+		baseUrl:      "https://quay.io",
+		numberOftags: cfg.NumberOfTags,
+	}
+	return q
+}
```

Comment:
What do you think about mounting our pull secrets into standard locations https://github.com/containers/image/blob/main/docs/containers-auth.json.5.md#description to avoid having to handle referencing/storing the secrets in our code?

Reply:
That's great. I'll remove the source auth and replace it with this. It will also implicitly enable auth for other registries.